### PR TITLE
Defines the direction in our Directed Acyclic Graph

### PR DESCRIPTION
**Pull request by kaushikcfd**

### CODE REVIEW

File: doc/design.rst [hunk 1]
```diff
@@ -81,6 +81,13 @@ Computation and Results
     Operations that *can* be expressed as :class:`~pytato.array.IndexLambda`
     without loss of information, *should* be expressed that way.
 
+- Every array node denotes a computation graph. In :mod:`pytato` the
+  computation graphs are always Directed Acyclic Graphs (DAG). This is a direct
+  consequence of an :class:`~pytato.array.Array` instance being immutable. We
+  choose the direction of the edges in the DAG to resemble the one typically
+  seen in a data flow graph, i.e. the successors of a node are its *users* and
+  the predecessors of a node are the arrays that it *uses*.
+
```
Comment (matthiasdiener) [inline doc/design.rst:90]: The items above have three spaces after the dash.

File: doc/design.rst [hunk 2]
```diff
@@ -81,6 +81,13 @@ Computation and Results
     Operations that *can* be expressed as :class:`~pytato.array.IndexLambda`
     without loss of information, *should* be expressed that way.
 
+- Every array node denotes a computation graph. In :mod:`pytato` the
+  computation graphs are always Directed Acyclic Graphs (DAG). This is a direct
```
Comment (matthiasdiener) [inline doc/design.rst:85]:
```suggestion
  computation graphs are always Directed Acyclic Graphs (DAGs). This is a direct
```

File: doc/design.rst (hunk 1 above)
Comment (matthiasdiener) [inline doc/design.rst:90]: Maybe it's worthwhile to add something like this:

```suggestion

  Note that although DAGs are usually visualized with edges from nodes to its users, they are usually traversed
  in the opposite direction, that is, going from a node to the the nodes that it uses.
```

**Review by matthiasdiener — COMMENTED** [review]
This looks like a great addition to the manual!

File: doc/design.rst (hunk 2 above)
Comment (kaushikcfd) [reply doc/design.rst]: Thanks!

File: doc/design.rst (hunk 1 above)
Comment (kaushikcfd) [reply doc/design.rst]: Thanks, fixed!

Comment (kaushikcfd) [reply doc/design.rst:96]: Thanks, added as a "gotcha" in 116e4b2.

**Review by matthiasdiener — APPROVED** [review]
LGTM, thanks!

**Comment by inducer** [comment]
Thanks!

**Pull request merged**